lib/ansible/modules/cloud/amazon/aws_organizations_facts.py

```diff
@@ -0,0 +1,151 @@
+        type: bool
+        required: false
+        default: 'no'
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather facts about the organization
+  aws_organizations_facts:
+
+- name: Facts are published in ansible_facts['organizations']
+  debug:
+    msg: "{{ ansible_facts['organizations'] }}"
+
+- name: Include a list of accounts in the AWS Organization
+  aws_organizations_facts:
+    accounts: true
+
+- name: Facts are published in ansible_facts['organizations']
+  debug:
+    msg: "{{ ansible_facts['organizations'] }}"
+'''
+
+RETURN = r'''
+organizations:
+  description: AWS Organization Facts.
+  returned: success
+  type: dict
+  sample:
+    id: o-abcde01234
+    arn: arn:aws:organizations::123456789012:organization/o-abcde01234
+    feature_set: ALL
+    master_account_arn: arn:aws:organizations::123456789012:account/o-abcde01234/123456789012
+    master_account_id: 123456789012
+    master_account_email: mailbox@example.com
+    available_policy_types:
+      - Type: SERVICE_CONTROL_POLICY
+        Status: ENABLED
+    accounts:
+      returned: success
+      type: list
+      sample:
+        - arn: arn:aws:organizations::210987654321:account/o-abcde01234/210987654321
+          email: mailbox+account@example.com
+          id: '210987654321'
+          joined_method: CREATED
+          joined_timestamp: 2019-01-01 16:33:33.333000
+          name: Another AWS Account
+          status: ACTIVE
+        - arn: arn:aws:organizations::123456789012:account/o-abcde01234/123456789012
+          email: mailbox@example.com
+          id: '123456789012'
+          joined_method: CREATED
+          joined_timestamp: 2018-01-01 15:22:22.222000
+          name: Master AWS Account
+          status: ACTIVE
+'''
+from ansible.module_utils.aws.core import (AnsibleAWSModule, HAS_BOTO3)
+from ansible.module_utils.ec2 import (AWSRetry, camel_dict_to_snake_dict)
+try:
+    from botocore.exceptions import (BotoCoreError, ClientError)
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def describe_organization(client, aws_retry=True):
+    return client.describe_organization(aws_retry=aws_retry)['Organization']
+
+
+def list_accounts(client):
+    paginator = client.get_paginator('list_accounts')
+    return paginator.paginate().build_full_result()['Accounts']
+
+
+def main():
+    argument_spec = dict(all_facts=dict(type='bool',
+                                        required=False,
+                                        default=False),
+                         accounts=dict(type='bool',
+                                       required=False,
+                                       default=False))
+    module = AnsibleAWSModule(argument_spec=argument_spec,
+                              supports_check_mode=True)
+    if not HAS_BOTO3:
```

Comment:
This is unnecessary; AnsibleAWSModule will already have terminated module execution if boto3 is not available.